test/integration/node-specific/resource_tracking_script_builder.ts

```diff
@@ -106,3 +133,68 @@ export async function runScript(
     heap
   };
 }
+
+/**
+ * A helper for running arbitrary MongoDB Driver scripting code in a resource information collecting script.
+ * This script uses info from node:process to collect resource information.
+ *
+ * **The provided function is run in an isolated Node.js process**
+ *
+ * A user of this function will likely need to familiarize themselves with the surrounding scripting, but briefly:
+ * - Every MongoClient you construct should have an asyncResource attached to it like so:
```

Comment:
Is this true?  I don't see us attaching async resources to mongo clients in your test functions.  If it is true, why?

Reply:
I've updated this with more accurate information!